### PR TITLE
{lib}[GCCcore/11.3.0] PortMidi v2.0.4, SDL2_image v2.6.3, SDL2_mixer v2.6.3, ... w/ Python 3.10.4

### PR DESCRIPTION
**Pull request by WilleBell**
(created using `eb --new-pr`)

### CODE REVIEW

**Comment by bedroge** [comment]
Test report by @bedroge
**FAILED**
Build succeeded for 4 out of 5 (5 easyconfigs in total)
interactive2 - Linux Rocky Linux 8.7, x86_64, AMD EPYC-Milan Processor (zen2), Python 3.6.8
See https://gist.github.com/bedroge/c1d736fa855147fee2b43def567b81a3 for a full test report.

**Comment by bedroge** [comment]
```
      Hunting dependencies...
      SDL     : found 2.0.22
      FONT    : found
      IMAGE   : found
      MIXER   : found
      PNG     : not found
```
Looks like this needs `libpng`?

File: easybuild/easyconfigs/p/pygame/pygame-2.1.0-GCCcore-11.3.0.eb [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+]
+
+use_pip = True
+
+_pygame_extra_base = ':'.join([
+    '$EBROOTSDL2_IMAGE',
+    '$EBROOTSDL2_MIXER',
+    '$EBROOTSDL2_TTF',
+    '$EBROOTLIBJPEGMINTURBO',
+    '$EBROOTPORTMIDI',
```
Comment (bedroge) [inline easybuild/easyconfigs/p/pygame/pygame-2.1.0-GCCcore-11.3.0.eb:35]: Adding the dependency on `libpng` didn't solve the error for me, but apparently it also has to be added here:

```suggestion
    '$EBROOTLIBPNG',
    '$EBROOTPORTMIDI',
```

**Comment by lexming** [comment]
@boegelbot: please test @ generoso

**Comment by lexming** [comment]
@boegelbot please test @ jsc-zen2

**Review by lexming — APPROVED** [review]
LGTM

**Comment by lexming** [comment]
@bedroge is is good on your side?

**Comment by bedroge** [comment]
Test report by @bedroge
**FAILED**
Build succeeded for 4 out of 5 (5 easyconfigs in total)
bob-Latitude-5300 - Linux Ubuntu 22.04, x86_64, Intel(R) Core(TM) i7-8665U CPU @ 1.90GHz, Python 3.10.6
See https://gist.github.com/bedroge/60fef8b0fe544d41b4e2018690bbd3a2 for a full test report.

**Comment by bedroge** [comment]
Test report by @bedroge
**SUCCESS**
Build succeeded for 5 out of 5 (5 easyconfigs in total)
interactive2 - Linux Rocky Linux 8.7, x86_64, AMD EPYC-Milan Processor (zen2), Python 3.6.8
See https://gist.github.com/bedroge/9bd7a68716af66517e47866e73bd5180 for a full test report.

**Comment by bedroge** [comment]
Now it does work on my Rocky machine, but it fails on my Ubuntu system:
Now I'm getting the following error:
```
      /data/apps/software/binutils/2.38-GCCcore-11.3.0/bin/ld: cannot find -lporttime: No such file or directory
```
I don't understand what should provide that library though (or why it perhaps only needs it on Ubuntu/Debian?), and I can't find it anywhere on my Rocky system either...

**Comment by bedroge** [comment]
Ah, yes, it depends on the distro, apparently:
https://github.com/pygame/pygame/blob/main/buildconfig/config_unix.py#L177

But I guess our EasyBuild-provided `PortMidi` should include everything (?), so maybe we can leverage that environment variable that they use there. Let me try that...

**Comment by branfosj** [comment]
From https://github.com/PortMidi/portmidi/issues/13

> Yes libporttime is merged into portmidi -- I don't know of anyone using it separately. Similarly, portmidi and porttime are duplicated in pmjni so Java programs only have to deal with one external library.

If I understand the [Debian/Ubuntu information](https://packages.ubuntu.com/focal/libportmidi-dev) then they added a patch to continue building `libporttime`.

File: easybuild/easyconfigs/p/pygame/pygame-2.1.0-GCCcore-11.3.0.eb [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+    '$EBROOTSDL2_MIXER',
+    '$EBROOTSDL2_TTF',
+    '$EBROOTLIBJPEGMINTURBO',
+    '$EBROOTLIBPNG',
+    '$EBROOTPORTMIDI',
+])
+
+exts_list = [
+    ('pygame', '2.1.0', {
+        'preinstallopts': 'PYGAME_EXTRA_BASE="%s"' % _pygame_extra_base,
```
Comment (bedroge) [inline easybuild/easyconfigs/p/pygame/pygame-2.1.0-GCCcore-11.3.0.eb:41]: Okay, that indeed seems to work, so we would need the following to make sure this works on Debian/Ubuntu (and maybe other distros):

```suggestion
        'preinstallopts': 'PORTMIDI_INC_PORTTIME=1 PYGAME_EXTRA_BASE="%s"' % _pygame_extra_base,
```

**Comment by bedroge** [comment]
@boegelbot please test @ generoso

**Comment by bedroge** [comment]
@boegelbot please test @ jsc-zen2

**Comment by bedroge** [comment]
Test report by @bedroge
**SUCCESS**
Build succeeded for 5 out of 5 (5 easyconfigs in total)
bob-Latitude-5300 - Linux Ubuntu 22.04, x86_64, Intel(R) Core(TM) i7-8665U CPU @ 1.90GHz, Python 3.10.6
See https://gist.github.com/bedroge/4cd646ec7e888bb020c8f7e5a5e559ef for a full test report.

**Comment by bedroge** [comment]
Test report by @bedroge
**SUCCESS**
Build succeeded for 5 out of 5 (5 easyconfigs in total)
interactive2 - Linux Rocky Linux 8.7, x86_64, AMD EPYC-Milan Processor (zen2), Python 3.6.8
See https://gist.github.com/bedroge/84f121a1fce2eb43b01ff2f28f65ac3d for a full test report.

**Review by bedroge — APPROVED** [review]
Lgtm

**Comment by bedroge** [comment]
Going in, thanks @WilleBell!

**Pull request merged**